config/bidderinfo.go

```diff
@@ -177,94 +166,367 @@ type SyncerEndpoint struct {
 	UserMacro string `yaml:"userMacro" mapstructure:"user_macro"`
 }
 
-// Override returns a new SyncerEndpoint object where values in the original are replaced by non-empty/non-default
-// values in the override. No changes are made to the original or override SyncerEndpoint.
-func (s *SyncerEndpoint) Override(original *SyncerEndpoint) *SyncerEndpoint {
-	if s == nil && original == nil {
-		return nil
+func ProcessBidderInfos(path string, configBidderInfos BidderInfos) (BidderInfos, []error) {
+	errs := make([]error, 0)
+	bidderInfos, err := LoadBidderInfoFromDisk(path)
+	if err != nil {
+		return nil, append(errs, fmt.Errorf("Unable to load bidderconfigs %v", err))
+	}
+	if len(configBidderInfos) > 0 {
+		bidderInfos, err = applyBidderInfoConfigOverrides(configBidderInfos, bidderInfos)
+		if err != nil {
+			return nil, append(errs, err)
+		}
 	}
 
-	var copy SyncerEndpoint
-	if original != nil {
-		copy = *original
+	errs = validateBidderInfos(bidderInfos)
+
+	return bidderInfos, errs
+}
+
+// LoadBidderInfoFromDisk parses all static/bidder-info/{bidder}.yaml files from the file system.
+func LoadBidderInfoFromDisk(path string) (BidderInfos, error) {
+	return loadBidderInfo(path)
+}
+
```

Comment:
Why do we have `LoadBidderInfoFromDisk(path)` when `loadBidderInfo(path)` is also necessarily from disk? I could see the utility if `loadBidderInfo()` was more general in some sense, but I don't see what we gain from it in this instance.

Reply:
Per my understanding it was done specifically for unit tests to mock data reading from file system. Implementations for "infoReader" are `infoReaderFromDisk` and `fakeInfoReader`. 

Do we want to support other types of data loading besides file system?

Reply:
My confusion rests more in that if `LoadBidderInfoFromDisk()` is an `infoReader` interface, then `loadBidderInfo()` is as well, as they have the same signature. Normally when I see a pattern like this, it is of the form:
```f1(A) X {
  return f2(A,B)
  }```
  Where B is some constant, and `f2` is some more general function than `f1` is supposed to be. Such as if `loadBidderInfo()` is intended to support other sources than disk some day, but where `LoadBidderInforFromDisk()` is called, will only ever be loading from disk. Using the pattern you have used here would allow for expanding `loadBidderInfo()` in the future without having to update the parts of code that currently call it. Since I am not aware of any plans to support other data sources, and `loadBidderInfo` is also an implementation of `infoReader`, I still do not understand why both functions exist like this.